File: Bitwise/A2.cpp
#include<iostream>
using namespace std;
typedef unsigned int UINT;
class Bitwise
{
public:
	
	UINT iValue1;
	int iPos;
	int iMask;

	Bitwise()
	{

	 	iValue1 = 0;
		iPos = 0;
		iMask = 0x00000001;
	}

	void Accept()
	{
		cout<<"Enter the Number:\n";
		cin>>iValue1;
	}	

};

class Marvellous:public Bitwise
{ 
public:


	UINT ONbit()
	{	
		int i = 0;
		int iCnt = 0;
		int iResult =0;
		 
	/*	for(i = 1;i<32;i++)
		{
				iResult = iValue1&iMask;

			if(iResult != 0)
			{
				iCnt++;

			}

			iMask = iMask<<1;

		} */

		for(i = 1;i<32;i++)
		{

			if((iValue1&iMask)==iMask)
			{

				iCnt++;
			}
			iMask = iMask<<1;

		}


		return iCnt;

	} 

};

int main()
{
	int iRet = 0;

	Marvellous mobj;

	mobj.Accept();
	
	iRet =mobj.ONbit();

	cout<<" No of On bits are :"<<iRet<<"\n";
	return 0;
}

File: File Handling/A2.java
import java.util.*;

class A2
{

	public static void main(String[] args) 
	{

		boolean bret = false;
		Marvellous mobj = new Marvellous();
		mobj.Accept();
		bret = mobj.CheckBit();

		if(bret == true)
		{

			System.out.println("TRUE");

		}
		else
		{
			System.out.println("False");
		}



	}
}


class Accept
{
	public int iValue1= 0;
	public int iValue2 = 0;

		void Accept()
		{
		
		Scanner sobj = new Scanner(System.in);
		
		System.out.println("Enter the number:");
		iValue1 = sobj.nextInt();


		System.out.println("Enter the number:");
		iValue2 = sobj.nextInt();
		
		}


}

class Marvellous extends Accept
{
	boolean CheckBit()
	{
		int iMask = 0x00000001;
		int iAns = 0;

		iMask = iMask<<(iValue2 -1);

		iAns = iValue1&iMask;

		if(iAns == iMask)
		{

			return true;
		}

		else
		
		{

			 return false;
		
		}

		

	}


}
